### ISSUE TITLE
`runner.batch_metrics` UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach()

### ISSUE THREAD

**Issue by km4sh** [catalyst-team/catalyst#1222]
## 🐛 Bug Report

While `runner.train()`, a **UserWarning** is raised:
```
<path>/python3.8/site-packages/catalyst/core/runner.py:716:
UserWarning: 
To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  k: runner.engine.sync_tensor(torch.tensor(v, device=runner.device), "mean")
```
I know that is because the `runner.batch_metrics['loss']` is a tensor, and the warning will be fixed with a float loss value.
However, a float loss value is not able to do BP. How to fix it properly?

### How To Reproduce
1. Train a runner
2. Watch the output from python

 

#### Code sample

```python
# my custom runner
class CustomRunner(dl.SupervisedRunner):
    def predict_batch(self, batch, **kwargs):
        return self.model(batch['mixture'])

    def handle_batch(self, batch):
        estimated = self.model(batch['mixture'])
        self.batch = {
            "mixture": batch['mixture'],
            "estimated": estimated,
            "matrix": torch.view_as_real(batch['matrix']).type(batch['mixture'].type())
        }
```
```python
# runner.batch_metrics
defaultdict(None,
            {'loss': tensor(0.5958, grad_fn=<MseLossBackward>),
             'lr': 0.02,
             'momentum': 0.9})
```

### Expected behavior

No warning.

### Environment
```bash
Catalyst version: 21.04.2
PyTorch version: 1.8.0
Is debug build: No
CUDA used to build PyTorch: None
TensorFlow version: N/A
TensorBoard version: N/A

OS: Mac OSX 11.3.1
GCC version: Could not collect
CMake version: version 3.19.6

Python version: 3.8
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] catalyst==21.4.2
[pip3] numpy==1.20.1
[pip3] tensorboardX==2.2
[pip3] torch==1.8.0
[pip3] torchaudio==0.8.0a0+a751e1d
[pip3] torchvision==0.2.2
[conda] blas                      2.108                       mkl    conda-forge
[conda] blas-devel                3.9.0                     8_mkl    conda-forge
[conda] catalyst                  21.4.2                   pypi_0    pypi
[conda] libblas                   3.9.0                     8_mkl    conda-forge
[conda] libcblas                  3.9.0                     8_mkl    conda-forge
[conda] liblapack                 3.9.0                     8_mkl    conda-forge
[conda] liblapacke                3.9.0                     8_mkl    conda-forge
[conda] mkl                       2020.4             h08c4f10_301    conda-forge
[conda] mkl-devel                 2020.4             h694c41f_302    conda-forge
[conda] mkl-include               2020.4             h08c4f10_301    conda-forge
[conda] numpy                     1.20.1           py38h64deac9_0    conda-forge
[conda] pytorch                   1.8.0                   py3.8_0    pytorch
[conda] tensorboardx              2.2                      pypi_0    pypi
[conda] torchaudio                0.8.0                      py38    pytorch
[conda] torchvision               0.2.2                      py_3    pytorch
```

### Checklist
- [x] bug description
- [x] steps to reproduce
- [x] expected behavior
- [x] environment
- [x] code sample / screenshots
  
### FAQ
Please review the FAQ before submitting an issue:
- [x] I have read the [documentation and FAQ](https://catalyst-team.github.io/catalyst/)
- [x] I have reviewed the [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [x] I have checked the [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [x] I have read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [x] I have joined [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

**Comment by Scitator** [comment]
Hi @km4sh ,
yeah, I know about the warning, nevertheless not sure if it required to be solved.
as a simple hotfix, you could use [warnings.filter](https://github.com/catalyst-team/catalyst/blob/master/catalyst/__init__.py#L4)
